src/net.md

```diff
@@ -876,6 +877,99 @@ fn run() -> Result<()> {
 # quick_main!(run);
 ```
 
+[ex-check-broken-links]: #ex-check-broken-links
+<a name="ex-check-broken-links"/>
+## Check a webpage for broken links
+
+[![reqwest-badge]][reqwest] [![select-badge]][select] [![url-badge]][url] [![cat-net-badge]][cat-net]
+
+We parse the URL using [`Url::parse`] and then retrieve the document using reqwest.
+
+We call "get_base_url" to retrieve the base URL. If the document has a "base" tag, we get the "href" [`attr`] from the first occurrence of the "base" tag. This is then used as the base URL. Otherwise, we can use [`Position::BeforePath`] with the original URL to get the base of that URL.
+
+We then retrieve all the links from the document, and [`collect`] them into a [`HashSet`] to remove any duplicates.
+
+Now that we have our URL, we pass them into the "get_broken_links" function along with the base URL. This will iterate over the URL, use [`url::ParseOptions`] and [`Url::parse`] for each to get the absolute URL and then perform a HTTP GET request. If the response status is 404, the URL will be added to the HashSet which is returned by the function.
+
+Finally, we check the contents of the HashSet. If it is empty, we report that no links are broken, otherwise we iterate over the HashSet and report the broken links.
+
+```rust,no_run
+# #[macro_use]
+# extern crate error_chain;
+extern crate reqwest;
+extern crate select;
+extern crate url;
+
+use url::{Url, Position};
+use reqwest::StatusCode;
+use select::document::Document;
+use select::predicate::Name;
+
+use std::collections::HashSet;
+#
+# error_chain! {
+#   foreign_links {
+#       ReqError(reqwest::Error);
+#       IoError(std::io::Error);
+#       UrlParseError(url::ParseError);
+#   }
+# }
+fn get_base_url(url: &Url, doc: &Document) -> Result<Url> {
+    let base_tag_href = doc.find(Name("base")).filter_map(|n| n.attr("href")).nth(0);
+
+    let base_url = base_tag_href.map_or_else(|| Url::parse(&url[..Position::BeforePath]), Url::parse)?;
+    
+    Ok(base_url)
+}
+
+fn get_broken_links<'a>(base_url: &Url, links: &HashSet<&'a str>) -> Result<HashSet<&'a str>> {
+    let mut broken_links = HashSet::new();
```

Comment:
I would for this function to operate only on a single `Url`  `check_link`, it would take only a single `Url` (or better yet `AsRef<Url>`) that was already made absolute (making the `base_url` param unnecessary). Then such function could be `map`'ed over the collection iterator instead of collected into yet another `HashSet`